server/src/managers/account_manager.rs

```diff
@@ -39,7 +41,22 @@ where
             phone_number,
             account_attributes,
         );
-        self.db.add_account(&account).await?;
+        self.db.add_account(&account).await.map_err(|err| {
+            let mut error = ApiError {
+                status_code: StatusCode::INTERNAL_SERVER_ERROR,
+                message: "Could not create account".into(),
+            };
+            match err.downcast_ref() {
+                Some(sqlx::Error::Database(err))
+                    if (err.as_ref()).constraint() == Some("phone_number") =>
+                {
+                    error.status_code = StatusCode::BAD_REQUEST;
+                    error.message += ", phone number already in use".into();
+                }
+                _ => (),
+            };
```

Comment:
```suggestion
            if let Some(sqlx::Error::Database(err)) = err.downcast_ref() {
                if (err.as_ref()).constraint() == Some("phone_number") =>
                {
                    error.status_code = StatusCode::BAD_REQUEST;
                    error.message += ", phone number already in use".into();
                }
            };
```
Dont know if it works, but looks better